server/routes/explore/helpers.py

```diff
@@ -450,4 +451,23 @@ def abort(error_message: str,
 def _set_blocked(err_json: Dict):
   err_json['blocked'] = True
   if err_json.get('debug'):
-    err_json['debug']['blocked'] = True
\ No newline at end of file
+    err_json['debug']['blocked'] = True
+
+
+def explore_post_body_cache_key():
+  """
+  Builds flask cache key for /detect and /detect-and-fulfill POST
+  requests. Does not cache requests with a contextHistory.
+  """
+  body_object = request.get_json()
+  if "contextHistory" in body_object and body_object["contextHistory"]:
+    # Don't cache requests with a contextHistory by returning None.
+    # All context histories are unique due to a unique session ID, so caching
+    # isn't helpful.
+    # TODO: Strip out session id, debug info, and potentially other fields from
+    # contextHistory so it can be cached
+    return None
```

Comment:
I see, so if the cache key is None, flask-caching will do nothing even using the default url path?

Reply:
yeah that's right